reference-implementation/lib/writable-stream.js

```diff
@@ -332,15 +315,16 @@ function WritableStreamHandleAbortRequestIfPending(stream) {
     return;
   }
 
-  WritableStreamFinishAbort(stream);
-
   const abortRequest = stream._pendingAbortRequest;
   stream._pendingAbortRequest = undefined;
   const promise = WritableStreamDefaultControllerAbort(stream._writableStreamController, abortRequest._reason);
   promise.then(
     abortRequest._resolve,
     abortRequest._reject
   );
+  stream._backpressure = false;
```

Comment:
I was just reaching the same conclusion.

Can we pass a boolean into WritableStreamError called messWithReadyPromise or similar?